### ISSUE TITLE
Blazor app doesn't load after upgrade to 3.0-preview4/preview5

### ISSUE THREAD

**Issue by vova-lantsov-dev** [aspnet/AspNetCore#10578]
I have latest extension installed from [here](https://github.com/aspnet/Blazor/releases/tag/3.0.0-preview4). Also VS 2019 16.2.0 Preview 1.0 installed. Code migrated from 0.7.0 using official migration guide.

My app can't load. Only router exists inside `<app>`.
![image](https://user-images.githubusercontent.com/45858087/58470373-39469200-814a-11e9-85dc-d84fa360096b.png)

Nothing useful in console
![image](https://user-images.githubusercontent.com/45858087/58470445-67c46d00-814a-11e9-83c0-5f58c0e66d13.png)

Debugging gives me no information at all
![image](https://user-images.githubusercontent.com/45858087/58470546-93dfee00-814a-11e9-9a31-aff3a9fe038d.png)

My project is available [here](https://github.com/vova-lantsov-dev/bookkeeping-blazor/tree/feature-blazor-3.0.0-preview4) (branch called feature-blazor-3.0.0-preview4)

Please help me determine what goes wrong. I added some logging inside OnInit(Async) methods of my pages, but they're never called. I tried to play with Startup file of Server project, but nothing changed.

**dotnet --info output:**
```
Пакет SDK для .NET Core (отражающий любой global.json):
 Version:   3.0.100-preview5-011568
 Commit:    b487ff10aa

Среда выполнения:
 OS Name:     Windows
 OS Version:  10.0.17134
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.0.100-preview5-011568\

Host (useful for support):
  Version: 3.0.0-preview5-27626-15
  Commit:  61f30f5a23

.NET Core SDKs installed:
  2.1.202 [C:\Program Files\dotnet\sdk]
  2.1.700 [C:\Program Files\dotnet\sdk]
  2.1.800-preview-009677 [C:\Program Files\dotnet\sdk]
  2.2.300 [C:\Program Files\dotnet\sdk]
  3.0.100-preview5-011568 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0-preview5-19227-01 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.0-preview5-27626-15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.0-preview5-27626-15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

**Comment by 3x0dv5** [comment]
looks like you project has a few things that you might need to fix.

Couple of things I did:
![image](https://user-images.githubusercontent.com/250271/58594890-d8f94280-8266-11e9-8483-b2313a230ced.png)

Maybe it would be a good idea to put a _Imports.razor in the root folder next to the App.razor with the minimum contents like:
```csharp
@using Microsoft.AspNetCore.Components
@using Microsoft.AspNetCore.Components.Routing
```
you entry page seems to be referring to a class that doesn't seem to exist
![image](https://user-images.githubusercontent.com/250271/58595062-5cb32f00-8267-11e9-8d48-82406c444eaf.png)

then in the browser's console you can see a bunch of errors
![image](https://user-images.githubusercontent.com/250271/58595238-e8c55680-8267-11e9-8c6c-46343810f62f.png)

That file BookkeepingNasheDetstvo.Client.Components.ForButtons.Button.razor, try call it something else. Something that doesn't conflict with an html tag when converted to lowercase, not sure if this is a known issue in Razor, I actually always call new components something like MyPrefixComponent.razor

In your `InputDateComponent.razor` you have a `<Button>` and that is causing a lot of the errors in the browser, I commented the line and was able to see stuff.
![image](https://user-images.githubusercontent.com/250271/58595551-f3ccb680-8268-11e9-904b-7d487cc1a49d.png)


I managed to see something on the screen.
![image](https://user-images.githubusercontent.com/250271/58595475-b23c0b80-8268-11e9-8f09-6f136a4acb35.png)

Anyways, hope this helps you in any way.

**Comment by 3x0dv5** [comment]
Actually ignore the `SubjectModal` part, that seems to be a bug in my Resharper.
Although you might be missing the `@inherits` in the `SubjectModal.razor`
![image](https://user-images.githubusercontent.com/250271/58595909-1a3f2180-826a-11e9-83a1-19e3a9f7ed93.png)

But you need an if check in your `subjects.razor`.
![image](https://user-images.githubusercontent.com/250271/58595977-4b1f5680-826a-11e9-8250-bfbb26671712.png)

**Comment by danroth27** [comment]
Thanks @3x0dv5 for helping out here!

@vova-lantsov-dev I'm hoping the pointers from @3x0dv5 are enough to get you unblocked. If you're still hitting issues that you think are problems we need to address in Blazor please let us know!

**Comment by vova-lantsov-dev** [comment]
@3x0dv5 for `subjects.razor` there is `@inherits` in _Imports.razor

After changing .csproj file to what you showed it still does not work
![image](https://user-images.githubusercontent.com/45858087/58703299-e0c4fe00-83b0-11e9-8c54-1472bdb68367.png)

Application is running well but not responding at all.
![image](https://user-images.githubusercontent.com/45858087/58703403-41543b00-83b1-11e9-8178-10d6bee95a9d.png)

I also commented `DateInputComponent`

**Comment by vova-lantsov-dev** [comment]
Can you please create PR to my repo so I can try to make things work?

**Comment by vova-lantsov-dev** [comment]
@3x0dv5 and also tell me your VS, .NET Core SDK and Blazor extension versions. I'd like to compare them with mine

**Comment by vova-lantsov-dev** [comment]
I found the same issue from 2018: https://github.com/aspnet/Blazor/issues/307

**Comment by vova-lantsov-dev** [comment]
Also some exception from Output tab
![image](https://user-images.githubusercontent.com/45858087/58704666-7793b980-83b5-11e9-83dc-67d69f01e5b8.png)

**Comment by vova-lantsov-dev** [comment]
@danroth27 , any suggestion? Is VS 2019 16.2.0 Preview 1.0 supported? According to issue I sent above, there should be a problem with Razor compiler.

**Comment by danroth27** [comment]
@vova-lantsov-dev Yes, VS2019 16.2 Preview 1 should be fine. Let me take a quick look at your repo.

**Comment by vova-lantsov-dev** [comment]
@danroth27 I can also give you remote access to my PC so you can take a look there

**Comment by danroth27** [comment]
@vova-lantsov-dev I sent you a PR that gets the initial page to render: https://github.com/vova-lantsov-dev/bookkeeping-blazor/pull/4. The tricky that had me stumped for a bit was that the initial page wasn't rendering because the namespace was missing for the Router component in App.razor. This problem will become easier to diagnose once we fix https://github.com/aspnet/AspNetCore/issues/9860. I'm sure there's more stuff to fix up, but hopefully this will get you unblocked.

**Comment by vova-lantsov-dev** [comment]
@danroth27 thank you, now it makes sense. I also know that there is an issue that components are not visible when usings are declared in _Layout.cshtml. Is it already reported or I should create new issue?

**Comment by danroth27** [comment]
> I also know that there is an issue that components are not visible when usings are declared in _Imports.razor . Is it already reported or I should create new issue?

That's expected. Layouts in Blazor are components defined by their own class. So using statements in the layout won't flow into individual pages. To propagate using statements across components you should use *_Imports.razor*.
